backend/uclapi/workspaces/occupeye.py

```diff
@@ -0,0 +1,348 @@
+                    "image_id": survey_map["ImageID"]
+                }
+            )
+            pipeline.rpush(
+                survey_maps_key,
+                survey_map["MapID"]
+            )
+            pipeline.expire(
+                survey_map_id,
+                self.SURVEY_TTL
+            )
+        pipeline.expire(
+            survey_maps_key,
+            self.SURVEY_TTL
+        )
+        pipeline.execute()
+
+    def _cache_survey_data(self):
+        """
+        This function will cache all surveys (e.g. buildings) in
+        the OccupEye system. It makes use of the _cache_maps_for_survey
+        helper function above to tie all maps to surveys.
+        """
+        # Use a Redis Pipeline to ensure that all the data is inserted together
+        # and atomically
+        pipeline = self.r.pipeline()
+        # Ensure that the list of occupeye surveys has actually been cleared
+        pipeline.delete("occupeye:surveys")
+
+        headers = {
+            "Authorization": self.get_bearer_token()
+        }
+        url = "{}/api/Surveys/?deployment={}".format(
+            self.base_url,
+            self.deployment_name
+        )
+        request = requests.get(
+            url=url,
+            headers=headers
+        )
+        surveys_data = request.json()
+
+        for survey in surveys_data:
+            survey_key = "occupeye:surveys:" + str(survey["SurveyID"])
+            pipeline.hmset(
+                survey_key,
+                {
+                    "id": survey["SurveyID"],
+                    "active": survey["Active"],
+                    "name": survey["Name"],
+                    "start_time": survey["StartTime"],
+                    "end_time": survey["EndTime"]
+                }
+            )
+            pipeline.expire(survey_key, self.SURVEY_TTL)
+            # We prepend to the list of Surveys because the API returns
+            # a list of surveys where the ID decrements
+            pipeline.lpush("occupeye:surveys", str(survey["SurveyID"]))
+            # Cache all maps for this survey for later use
+            self._cache_maps_for_survey(survey["SurveyID"])
+
+        pipeline.expire("occupeye:surveys", self.SURVEY_TTL)
+        pipeline.execute()
+
+    def get_surveys(self):
+        """
+        Serialises all surveys and maps to a dictionary
+        object that can be returned to the user. If the
+        requisite data does not exist in Redis, it is cached using
+        the helper functions above, then returned from the cache.
+        """
+        if self.r.llen("occupeye:surveys") == 0:
+            # The data is not in the cache, so cache it.
+            self._cache_survey_data()
+
+        # Now we know we have the data cached, we can serve it
+        survey_ids = self.r.lrange(
+            "occupeye:surveys",
+            0,
+            self.r.llen("occupeye:surveys") - 1
+        )
+
+        surveys = []
+
+        for id in survey_ids:

```

Comment:
`id` is a builtin function in python. Better rename this variable.

Reply:
good shout